src/connect/components/tree-view-item/tree-view-item.tsx

```diff
@@ -126,39 +138,120 @@ const getItemIcon = (type: ItemType) => {
     }
 };
 
-export function ConnectTreeViewItem<T extends string = DefaultOptionId>(
-    props: ConnectTreeViewItemProps<T>,
-) {
+export function ConnectTreeViewItem(props: ConnectTreeViewItemProps) {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [mouseIsWithinButtonContainer, setMouseIsWithinButtonContainer] =
+        useState(false);
+    const [showDropdownMenu, setShowDropdownMenu] = useState(false);
+
     const {
         item,
-        onClick,
         children,
+        buttonProps,
         onDropEvent,
         onOptionsClick,
-        level = 0,
-        buttonProps = {},
-        defaultOptions = DefaultOptions,
-        ...divProps
+        ...delegatedProps
     } = props;
 
-    const containerRef = useRef(null);
+    const { dragProps, dropProps, isDropTarget } = useDraggableTarget<TreeItem>(
+        {
+            data: item,
+            onDropEvent: props.onDropEvent,
+        },
+    );
+
+    const isWriting =
+        item.action === ActionType.New || item.action === ActionType.Update;
+    const isHighlighted = getIsHighlighted();
 
-    const { dragProps, dropProps, isDropTarget } = useDraggableTarget<
-        TreeItem<T>
-    >({
-        data: item,
-        onDropEvent,
-    });
+    function getButtonProps() {
+        const { className: buttonClassName, ...restButtonProps } =
+            buttonProps ?? {};
 
-    const { className: buttonClassName, ...restButtonProps } = buttonProps;
+        const onMouseEnter: MouseEventHandler = event => {
+            event.stopPropagation();
+            if (event.target === event?.currentTarget) {
+                setMouseIsWithinButtonContainer(true);
+            }
+        };
+
+        const onMouseLeave: MouseEventHandler = event => {
+            event.stopPropagation();
+            if (event.target === event?.currentTarget) {
+                setMouseIsWithinButtonContainer(false);
+            }
+        };
+
+        const className = twMerge(
+            'py-3 rounded-lg transition-colors',
+            getButtonBackgroundClass(),
+            typeof buttonClassName === 'string' && buttonClassName,
+        );
 
-    const optionsContent = (defaultOptions || item.options) &&
-        onOptionsClick && (
-            <ConnectDropdownMenu<T>
-                items={
-                    item.options ||
-                    (defaultOptions as ConnectDropdownMenuItem<T>[])
+        return {
+            ...restButtonProps,
+            onMouseEnter,
+            onMouseLeave,
+            className,
+            ref: containerRef,
+        };
+    }
+
+    function getIsHighlighted() {
```

Comment:
These cases determine if an item is highlighted. It would have been convenient to just use tailwind's hover detection, but unfortunately that will include the component's children as well.